mule-packager/src/main/java/org/mule/tools/api/packager/ProjectInformation.java

```diff
@@ -0,0 +1,116 @@
+package org.mule.tools.api.packager;
+/*
+ * Mule ESB Maven Tools
+ * <p>
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * <p>
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.nio.file.Path;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Represents the basic information of a project.
+ */
+public class ProjectInformation {
+
+  private String groupId;
+  private String artifactId;
+  private String version;
+  private String packaging;
+  private Path projectBaseFolder;
+  private Path buildDirectory;
+
+  public ProjectInformation(String groupId, String artifactId, String version, String packaging, Path projectBaseFolder,
+                            Path buildDirectory) {
+    this.groupId = groupId;
```

Comment:
Please make this constructor private to force users to use the builder, thus forcing the validations